src/renderer/backend/dx11.rs

```diff
@@ -612,12 +625,40 @@ impl TextureHeap {
         })?;
 
         let id = TextureId::from(self.textures.len());
-
-        let texture = Texture { resource, shader_resource_view, id };
-        self.textures.push(texture);
+        self.textures.push(Texture { resource, shader_resource_view, id, width, height });
 
         Ok(id)
     }
+
+    unsafe fn update_texture(
+        &mut self,
+        texture_id: TextureId,
+        data: &[u8],
+        width: u32,
+        height: u32,
+    ) -> Result<()> {
+        let texture = &mut self.textures[texture_id.id()];
+        if texture.width != width || texture.height != height {
+            return Err(Error::new(
```

Comment:
I'd rather not construct the errors this way: they call out to a function which is not always available and leads to crashes on Wine, check out #167. 

This is not ideal since it relies on logging, but I'd use the following idiom instead:

```rs
error!(format!(/* ... */));
return Err(Error::from_hresult(HRESULT(-1));
```

In the future I'd like to go with a custom error format but for now this'll do since it's used only in a few spots.